### PR TITLE
feat: ofbool_eq_iff_eq

### PR DESCRIPTION
**Pull request by bollu**
shows that `BitVec.ofBool` is injective.

### CODE REVIEW

File: Std/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -89,6 +89,9 @@ theorem eq_of_toFin_eq {x y : BitVec w} (w : x.toFin = y.toFin) : x = y := by
 theorem ofNat_one (n : Nat) : BitVec.ofNat 1 n = BitVec.ofBool (n % 2 = 1) :=  by
   rcases (Nat.mod_two_eq_zero_or_one n) with h | h <;> simp [h, BitVec.ofNat, Fin.ofNat']
 
+theorem BitVec.ofBool_eq_iff_eq (b b' : Bool) : BitVec.ofBool b = BitVec.ofBool b' ↔ b = b' := by
```
Comment (tobiasgrosser) [inline Std/Data/BitVec/Lemmas.lean:92]:
```suggestion
theorem ofBool_eq_iff_eq (b b' : Bool) : BitVec.ofBool b = BitVec.ofBool b' ↔ b = b' := by
```

**Comment by tobiasgrosser** [comment]
awaiting-review

File: Std/Data/BitVec/Lemmas.lean [hunk 2]
```diff
@@ -89,6 +89,9 @@ theorem eq_of_toFin_eq {x y : BitVec w} (w : x.toFin = y.toFin) : x = y := by
 theorem ofNat_one (n : Nat) : BitVec.ofNat 1 n = BitVec.ofBool (n % 2 = 1) :=  by
   rcases (Nat.mod_two_eq_zero_or_one n) with h | h <;> simp [h, BitVec.ofNat, Fin.ofNat']
 
+theorem ofBool_eq_iff_eq (b b' : Bool) : BitVec.ofBool b = BitVec.ofBool b' ↔ b = b' := by
+  constructor <;> intros h <;> cases b <;> cases b' <;> simp at h ⊢ <;> try contradiction
```
Comment (alexkeizer) [inline Std/Data/BitVec/Lemmas.lean:93]: You don't need the `constructor` tactic, `simp` is able to simplify trivial `iff`s pretty well. I also squeezed the non-terminal simp, negating some of the golfing:
```suggestion
  cases b <;> cases b'
    <;> simp only [ofBool_false, ofNat_eq_ofNat, ofBool_true, iff_false]
    <;> (intro; contradiction)
```

File: Std/Data/BitVec/Lemmas.lean [hunk 3]
```diff
@@ -89,6 +89,11 @@ theorem eq_of_toFin_eq {x y : BitVec w} (w : x.toFin = y.toFin) : x = y := by
 theorem ofNat_one (n : Nat) : BitVec.ofNat 1 n = BitVec.ofBool (n % 2 = 1) :=  by
   rcases (Nat.mod_two_eq_zero_or_one n) with h | h <;> simp [h, BitVec.ofNat, Fin.ofNat']
 
+theorem ofBool_eq_iff_eq (b b' : Bool) : BitVec.ofBool b = BitVec.ofBool b' ↔ b = b' := by
+  cases b <;> cases b'
+    <;> simp only [ofBool_false, ofNat_eq_ofNat, ofBool_true, iff_false]
+    <;> (intro; contradiction)
+
```
Comment (joehendrix) [inline Std/Data/BitVec/Lemmas.lean:96]: The kernel can prove this:

```suggestion
theorem ofBool_eq_iff_eq : ∀(b b' : Bool), BitVec.ofBool b = BitVec.ofBool b' ↔ b = b' := by
  decide

```

Comment (bollu) [reply Std/Data/BitVec/Lemmas.lean]: @joehendrix , suggestion accepted.

**Pull request merged**